### PR TITLE
New AbstractFSM

### PR DESCRIPTION
**Pull request by kanak8278**
[added Parser which will pick OptionParser or LLMManager](https://github.com/OpenNyAI/Jugalbandi-Manager/commit/0f887321c0659b7bffe5b471665bcba9b5aada28)

Wrote helper functions in AbstractFSM

Pydantic Variables Class for FSM Variables

### CODE REVIEW

**Comment by sameersegal** [comment]
@kanak8278 please remove extra logging steps

Please update the CarWash example

Please ensure this library gets built correctly as well

File: flow/pyproject.toml [hunk 1]
```diff
@@ -22,7 +22,7 @@ pydantic = "^2.5.3"
 azure-storage-blob = "^12.19.0"
 azure-identity = "^1.15.0"
 aiofiles = "^23.2.1"
-aiohttp = "^3.9.4"
+aiohttp = "^3.9.1"
```
Comment (sameersegal) [inline flow/pyproject.toml:25]: Don't downgrade poetry packages.

File: jb-manager-bot/pyproject.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "jb-manager-bot"
```
Comment (sameersegal) [inline jb-manager-bot/pyproject.toml:2]: Rename the library to jb-manager-app

**Review by sameersegal — CHANGES_REQUESTED** [review]

File: jb-manager-bot/pyproject.toml (hunk 2 above)
Comment (kanak8278) [reply jb-manager-bot/pyproject.toml:2]: In pypi the name of the library is jb-manager-bot. Checking if can be modifed.

**Comment by sameersegal** [comment]
Update this page as well: https://opennyai.github.io/Jugalbandi-Manager/how-tos/fsm_building.html

Bump the version to 0.2 because it a big change.

**Comment by shreypandey** [comment]
@kanak8278 Please take a pull from main and resolve conflicts.

**Comment by kanak8278** [comment]
> @kanak8278 Please take a pull from main and resolve conflicts.

Merged Conflicts settled

**Pull request closed without merging**